drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -709,6 +710,36 @@ static int siwx91x_stats(const struct device *dev, struct net_stats_wifi *stats)
 }
 #endif
 
+static int siwx91x_wifi_version(const struct device *dev, struct wifi_version *params)
+{
+	sl_wifi_firmware_version_t fw_version = { 0 };
+	struct siwx91x_dev *sidev = dev->data;
+	static char fw_version_str[128];
```

Comment:
This looks a bit excessive if we only print couple of ints. If we make this smaller, we could use a stack variable and save some memory.

Reply:
Indeed, 32 bytes seems sufficient. However, I don't believe we can allocate it on the stack since `params->fw_version` is a `char *`.